### PR TITLE
Keep headers after an error (#11) for v2.x

### PR DESCRIPTION
**Pull request by PlasmaPower**

### CODE REVIEW

**Comment by fengmk2** [comment]
eslint break.

**Comment by PlasmaPower** [comment]
Oops! I'll fix it in the morning. It feels like there should probably be an email for a breaking PR.

**Comment by PlasmaPower** [comment]
Okay lint is passing now, should be working on travis.

File: index.js [hunk 1]
```diff
@@ -92,29 +100,40 @@ module.exports = function(options) {
...
 
       ctx.status = 204;
     }
+
+    let promise = next();
+    if (options.keepHeadersOnError) {
+      promise = promise.catch(function(err) {
+        err.headers = err.headers || {};
+        Object.assign(err.headers, headersSet);
+        console.log(err.headers);
```
Comment (fengmk2) [inline index.js]: remove console.log

**Comment by PlasmaPower** [comment]
Oops! The log came from the cherry pick, not sure how that happened but it's not on master.

**Comment by fengmk2** [comment]
+1

**Comment by fengmk2** [comment]
2.1.0

**Pull request merged**